variables.tf

```diff
@@ -165,6 +165,7 @@ variable "node_pools" {
     node_count_min_per_zone     = number
     node_count_max_per_zone     = number
     node_labels                 = map(string)
+    node_taints                 = list(object({ key = string, value = string, effect = string }))
```

Comment:
Consider adding some doc (especially about the acceptable values of `effect` [link for ref](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#effect) ) along with the other variable descriptoins we have.